packages/app/src/systems/Account/hooks/useAccounts.tsx

```diff
@@ -49,12 +50,21 @@ const listenerAccountFetcher = () => {
 
 export function useAccounts() {
   const shouldListen = useRef(true);
+  const { assets } = useAssets();
   const hasBalance = store.useSelector(Services.accounts, selectors.hasBalance);
   const accountStatus = store.useSelector(Services.accounts, selectors.status);
   const ctx = store.useSelector(Services.accounts, selectors.context);
   const accounts = store.useSelector(Services.accounts, selectors.accounts);
   const account = store.useSelector(Services.accounts, selectors.account);
   const overlay = useOverlay();
+  const balanceAssets = useMemo(
+    () =>
+      account?.balances?.map((balance) => ({
+        ...balance,
+        ...(assets?.find(({ assetId }) => assetId === balance.assetId) || {}),
+      })),
+    [account?.balance, assets]
+  );
```

Comment:
Should we move this code to inside the machine?

Reply:
no because we would need to inject `assets` in machine context, it's not needed as it comes from `useAssets`.

I will put this code in a closure selector instead